### PR TITLE
Task5

### PR DESCRIPTION
**Pull request by petemarianne**

### CODE REVIEW

**Comment by petemarianne** [comment]
here's a lot of commits because for some reason extra files appeared on the branch

File: task5/classes/MyFilter.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public void doFilter (ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        long start = System.currentTimeMillis();
+        chain.doFilter(request, response);
+        long end = System.currentTimeMillis();
+
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        String method = httpRequest.getMethod();
+        String url = String.valueOf(httpRequest.getRequestURL());
+
+        System.out.println("\n" + method + " - " + url + " - " + (end - start) + "ms");
```
Comment (smiakchilo) [inline task5/classes/MyFilter.java]: Let's learn running cleaner code from the start)
`System.out.printf("Method: %s, url: %s, duration: %dms%n", method, url, end - start);`

This is the way to interpolate strings. `%s` means "a string argument from the ones specified in the following arguments list", `%d` means "numeric value", `%n` is for "new line"
Thus, no string concatenation.
Also, you missed "url" value in your code ;)

File: task5/classes/Name.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/get")
+public class Name extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String name = request.getParameter("name");
+        if (name.length()>100){
```
Comment (smiakchilo) [inline task5/classes/Name.java]: (cosmetic) Spaces around `>` is required for clean code

File: task5/classes/Page.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/page")
+public class Page extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        request.getRequestDispatcher("/WEB-INF/page.html").forward(request, response);
```
Comment (smiakchilo) [inline task5/classes/Page.java]: Also, from the clean code
It's required to  declare all "string literals" as constants
```
private static final String STARTING_PAGE_ADDRESS = "/WEB-INF/page.html";
//....
request.getRequestDispatcher(STARTING_PAGE_ADDRESS ) // ...
```
Constants are situated immediately at the beginning of a class

File: task5/classes/Status.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/status")
+public class Status extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.getWriter().write("<p style='color:red'>Application Is Running</p>");
```
Comment (smiakchilo) [inline task5/classes/Status.java]: The constants rule is universal, it is for all the literals and "magic numbers" anywhere they are used

**Comment by smiakchilo** [comment]
> here's a lot of commits because for some reason extra files appeared on the branch

Meaningful commit messages are *still* not there

File: task5/classes/MyFilter.java (hunk 1 above)
Comment (petemarianne) [reply task5/classes/MyFilter.java]: okay, i got it, but i don't understand what you mean by missing the "url" value... i literally used it in print.

**Comment by smiakchilo** [comment]
Uh, sorry my fault. You had the url. I mixed it up with a “label”

**Comment by petemarianne** [comment]
@smiakchilo, okay, thanks. so can I merge the branch or should I still change and correct something?

**Comment by smiakchilo** [comment]
Yes you can)

**Pull request merged**